lib/cloudwatchlogger/client/aws_sdk/threaded.rb

```diff
@@ -93,13 +93,11 @@ def deliver(message)
         end
 
         def connect!(opts = {})
-          @client = Aws::CloudWatchLogs::Client.new(
-            region: @opts[:region] || 'us-east-1',
-            access_key_id: @credentials[:access_key_id],
-            secret_access_key: @credentials[:secret_access_key],
-            http_open_timeout: opts[:open_timeout],
-            http_read_timeout: opts[:read_timeout]
-          )
+          args = { http_open_timeout: opts[:open_timeout], http_read_timeout: opts[:read_timeout] }
+          args[:region] = @opts[:region] if @opts[:region]
+          args.merge( @credentials.key?(:access_key_id) ? { access_key_id: @credentials[:access_key_id], secret_access_key: @credentials[:secret_access_key] } : {} )
```

Comment:
How does this work?

Reply:
The AWS SDK will look in the default provider chain for credentials if none are provided in the constructor, so here we check if the user has explicitly specified credentials and/or a region, and only provide them to the `CloudWatchLogs` client if so.